app/src/main/java/net/programmierecke/radiodroid2/FragmentTabs.java

```diff
@@ -114,57 +126,58 @@ private void setupViewPager(ViewPager viewPager) {
             if (countryCode != null) {
                 if (countryCode.length() == 2) {
                     Log.d("MAIN", "Found countrycode " + countryCode);
-                    adresses[0] = "json/stations/bycountrycodeexact/" + countryCode + "?order=clickcount&reverse=true";
+                    addresses[IDX_LOCAL] = "json/stations/bycountrycodeexact/" + countryCode + "?order=clickcount&reverse=true";
                 }else{
                     Log.e("MAIN", "countrycode length != 2");
                 }
             }else{
                 Log.e("MAIN", "device countrycode and sim countrycode are null");
             }
         }
-        for (int i = 0; i < fragments.length; i++) {
-            Bundle bundle = new Bundle();
-
-            if (i < 5) {
-                fragments[i] = new FragmentStations();
-            } else if (i < 8) {
-                fragments[i] = new FragmentCategories();
-            } else {
-                fragments[i] = new FragmentStations();
-                bundle.putBoolean(FragmentStations.KEY_SEARCH_ENABLED, true);
-            }
-
-            bundle.putString("url", adresses[i]);
-            fragments[i].setArguments(bundle);
+        fragments[IDX_LOCAL] = new FragmentStations();
+        fragments[IDX_TOP_CLICK] = new FragmentStations();
+        fragments[IDX_TOP_VOTE] = new FragmentStations();
+        fragments[IDX_CHANGED_LATELY] = new FragmentStations();
+        fragments[IDX_CURRENTLY_HEARD] = new FragmentStations();
+        fragments[IDX_TAGS] = new FragmentCategories();
+        fragments[IDX_COUNTRIES] = new FragmentCategories();
+        fragments[IDX_LANGUAGES] = new FragmentCategories();
+        fragments[IDX_SEARCH] = new FragmentStations();
+
+        for (int i=0;i<fragments.length;i++) {
+            Bundle bundle1 = new Bundle();
+            bundle1.putString("url", addresses[i]);
```

Comment:
something like the following seems to be missing here to make search work:
```
if (i == IDX_SEARCH) {
    bundle1.putBoolean(FragmentStations.KEY_SEARCH_ENABLED, true);
}
```

Reply:
good catch, applied now.